draft-ietf-scitt-architecture.md

```diff
@@ -551,64 +578,131 @@ Editor's Note: This may be moved to appendix.
+    - Body: `{ "operationId": "<Operation ID>", "status": "succeeded", "entryId": "<Entry ID>" }`
+
+- Status 200 - Registration failed
+    - Header `Content-Type: application/json`
+    - Body: `{ "operationId": "<Operation ID>", "status": "failed", "error": { "type": "<type>", "detail": "<detail>" } }`
+    - Error code: `badSignatureAlgorithm`
+    - Error code: `TBD`
+
+- Status 404 - Unknown Operation ID
+    - Error code: `operationNotFound`
+    - This can happen if the operation ID has expired and been deleted.
+
+If an operation failed, then error details SHOULD be embedded as a JSON problem details object in the `"error"` field.
+
+If an operation ID is invalid (i.e., it does not correspond to any submit operation), a service may return either a 404 or a `pending` status. This is because differentiating between the two may not be possible in an eventually consistent system.
+
+### Retrieve Registration Entry
 
 #### Request
 
 ~~~
-GET <Base URL>/entries/<Transaction ID>/receipt
+GET <Base URL>/entries/<Entry ID>
 ~~~
 
 #### Response
 
 One of the following:
 
-- HTTP Status `200` - Registration was successful and the Receipt is returned.
-- HTTP Status `400` - Transaction exists but does not correspond to a Registration Request.
-  - Error code `TransactionMismatch`
-- HTTP Status `404` - Transaction is pending, unknown, or invalid.
-  - Error code `TransactionPendingOrUnknown`
-  - Error code `TransactionInvalid`
+- Status 200.
+  - Header `Content-Type: application/json`
+  - Body `{ "entryId": "<Entry ID>" }`
+- Status 404 - Entry not found.
+  - Error code: `entryNotFound`
 
-The `200` response contains the SCITT_Receipt in the body.
+### Retrieve Registered Claim
 
-The `400` and `404` responses return the error details as described earlier.
+#### Request
 
-The retrieved Receipt may be embedded in the corresponding COSE_Sign1 document in the unprotected header, see TBD.
+~~~
+GET <Base URL>/entries/<Entry ID>/claim
+~~~
+
+Query parameters:
+
+- (Optional) `embedReceipt=true`
+
+If the query parameter `embedReceipt=true` is provided, then the claim is returned with the corresponding registration receipt embedded in the claim's COSE unprotected header.
+
+#### Response
+
+One of the following:
+
+- Status 200.
+  - Header: `Content-Type: application/cose`
+  - Body: COSE_Sign1 claim
 
-[TODO] There's also the `GET <Base URL>/entries/<Transaction ID>` endpoint which returns the submitted COSE_Sign1 with the Receipt already embedded. Is this useful?
+- Status 404 - Entry not found.
+  - Error code: `entryNotFound`
+
+### Retrieve Registration Receipt
+
+#### Request
+
+~~~
+GET <Base URL>/entries/<Entry ID>/receipt
+~~~
+
+#### Response
+
+One of the following:
+
+- Status 200.
+  - Header: `Content-Type: application/cbor`
+  - Body: SCITT_Receipt
+- Status 404 - Entry not found.
+  - Error code: `entryNotFound`
+
+The retrieved Receipt may be embedded in the corresponding COSE_Sign1 document in the unprotected header, see TBD.
```

Comment:
_"[...] see TBD"_ ?

Reply:
I added the reference to the receipts I-D.